accepted/2021/system-drawing-win-only/system-drawing-win-only.md

```diff
@@ -0,0 +1,183 @@
+  6, it's only supported on Windows, via the platform compatibility analyzer
+* Cross-platform consumers have the ability to turn on a compatibility switch to
+  continue using `System.Drawing.Common` via `libgdiplus`
+
+### Non-Goals
+
+* Closing the gap of `System.Drawing.Common` for non-Windows platforms or
+  bundling `libgdiplus` as part of the .NET 6 installer
+
+## Stakeholders and Reviewers
+
+* Libraries team
+* ML.NET
+* Xamarin team
+
+## Design
+
+In the past we decided to include the following disclaimer in our docs:
+
+>   The `System.Drawing.Common` NuGet package works on Windows, Linux, and macOS.
+>   However, there are some platform differences. On Linux and macOS, the GDI+
+>   functionality is implemented by the `libgdiplus` library. This library is not
+>   installed by default in most Linux distributions and doesn't support all the
+>   functionality of GDI+ on Windows on macOS. There are also platforms where
+>   `libgdiplus` is not available at all. To use types from the
+>   `System.Drawing.Common` package on Linux and macOS, you must install
+>   `libgdiplus` separately. For more information, see Install .NET on Linux or
+>   Install .NET on macOS.
+>
+>   If you can't use System.Drawing with your application, recommended
+>   alternatives include ImageSharp, SkiaSharp, and Windows Imaging Components.
+
+However, this disclaimer doesn't feel clear enough to steer people towards using
+other graphics libraries.
+
+The proposal is to do the following for .NET 6:
+
+1. **Mark the assembly as only being supported on Windows**. That will cause
+   the platform compatibility analyzer to warn when consumed from a project
+   that isn't targeting net6.0-windows (or marked as Windows-specific).
+
+2. Throw `PlatformNotSupportedException` when trying to load `libgdiplus`
+   pointing to an aka.ms link with the reasons behind it and recommending
+   stable, well-maintained open-source library alternatives
+
+3. Add an AppContext switch customers can set in their apps or at runtime to
+   re-enable `System.Drawing.Common` via `libgdiplus` on non-Windows platforms.
+
+4. Add targets to the package that when it is restored using a non-windows RID,
+   it warns that we no longer support it on non-windows and point to the aka.ms
+   link with more info.
+
+The challenge is that `System.Drawing.Common` is used by 1st party consumers
+such as ML.NET, which uses it for image manipulation and even exposes an API
+that returns a Bitmap. At the same time, `System.Devices.Iot` is in the process
+of moving to use ImageSharp and so far they had a great migration experience. We
+need to engage with 1st party customers to see what impact this proposal will
+have on them.
+
+***Note:** This only affect `System.Drawing.Common` and would leave
+`System.Drawing.Primitives` as-is. That assembly on contains primitive types
+that don't depend on GDI+, such `Rectangle`, `Point`, and `Size`.*
```

Comment:
Is there anything else in System.Drawing.Common that doesn't depend on libgdiplus and we should push down?

Reply:
I haven't seen any, but still doing an analysis to see if we need to push types down.